cvat/apps/engine/serializers.py

```diff
@@ -287,12 +287,15 @@
 
             logger.info("Label id {} ({}) was updated".format(db_label.id, db_label.name))
         else:
-            db_label = models.Label.objects.create(
-                name=validated_data.get('name'),
-                type=validated_data.get('type'),
-                parent=parent_label,
-                **parent_info
-            )
+            try:
+                db_label = models.Label.objects.create(
+                    name=validated_data.get('name'),
+                    type=validated_data.get('type'),
+                    parent=parent_label,
+                    **parent_info
+                )
+            except IntegrityError as exc:
+                raise exceptions.ValidationError(str(exc)) from exc
```

Comment:
@yasakova-anastasia, probably, now these warnings are actually valid. I suggest to check for the message and forward only if the message is safe. Please also remove or reuse the `InvalidLabel` exception. Another option is that we override the model create/save methods and add this try-catch to reraise `InvalidLabel`. This second variant can even provide better encapsulation.

Reply:
@zhiltsov-max, I overridden the `save` method and added the `create` method. I moved the try-catch logic to these methods. And now an error message about the uniqueness of labels is displayed instead of an `IntegrityError` message. Could you please review again?